### ISSUE TITLE
508-defect-2 [KEYBOARD, FOCUS MANAGEMENT]: Focus MUST be managed after evidence is uploaded

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#19688]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a keyboard user, I expect focus to be placed onto the first focusable field of my newly uploaded document.

## Details

Focus is not managed after a user has uploaded documentation on the supporting evidence page:
- Focus does not get set to the next logical step, document type
- Focus is sent to the top of the list loop pattern which causes issues when multiple uploads exist as the user will have to manually find where they left off
- JAWS/Chrome will announce to the user "File a Benefits Delivery at Discharge claim" and send them to the top of the page, on tab, the focus is sent to "add another document" which skips over the document type field

## Acceptance Criteria

- [ ] Once a user uploads a document, focus is sent to the document type field

## Environment

* Operating System: any
* Browser: any
* Screenreading device: any
* Server destination: production

## Steps to Recreate

1. Reach the supporting evidence page
2. Upload a document
3. Confirm focus is sent to the top of the list loop pattern

## Proposed Solution

Focus should logically be set to the document type select box as it is the next step in the upload process.

## Screenshots or Trace Logs
Upon uploading a document, focus is not placed onto the document that has been uploaded. Instead, it is sent to the start of the list loop.
https://user-images.githubusercontent.com/14154792/107420412-df852780-6ae6-11eb-9d50-0354d9ffc00a.mov

<img width="951" alt="Screen Shot 2021-02-09 at 3 02 32 PM" src="https://user-images.githubusercontent.com/14154792/107421417-efe9d200-6ae7-11eb-85aa-2039348f901f.png">

**Comment by Mottie** [comment]
Fix has been merged, and should be available in staging soon

**Comment by joshkimux** [comment]
@Mottie confirmed this is fixed! Focus is being moved to the document type.
Some minor things that may have to be addressed (defect level 3/4s)
- We may need to link the document type field to the document title/description using `aria-describedby`. That way, upon focus being moved to "Document type" the user will also hear "Document type, download.png, PTSD 781a form supporting documents." It would make navigating a list of upload fields much easier to follow since folks would only need to use the `tab` key as opposed to navigating line by line. I'm not sure if that would be plausible with the forms system, but it would be a solid improvement to the screen reader experience 😦 
- It appears that in the case a user _adds_ a file, triggers an error with that file, and then deletes it... and then adds a _new_ file, the new file will appear _along_ with the error from the deleted file. Am I interpreting that behavior correctly? And if so, is there any way to fix it? 

https://user-images.githubusercontent.com/14154792/113579991-0cd7e880-95f3-11eb-9edb-e3601c197898.mov

**Comment by joshkimux** [comment]
Closing out since the main issue has been fixed!
